server.go

```diff
@@ -3519,8 +3519,53 @@ type MessagingConn interface {
 
 // DataNode represents a data node in the cluster.
 type DataNode struct {
+	mu  sync.RWMutex
 	ID  uint64
 	URL *url.URL
+
+	// downCount is the number of times the DataNode has been marked as down
+	downCount uint
+
+	// OfflineUntil is the time when the DataNode will no longer be consider down
+	OfflineUntil time.Time
+}
+
+// Down marks the DataNode as offline for a period of time.  Each successive
+// call to Down will exponentially extend the offline time with a maximum
+// offline time of 5 minutes.
+func (d *DataNode) Down() {
+	d.mu.Lock()
+	defer d.mu.Unlock()
+
+	// Clamp the timeout to 5 mins max
+	t := 2 << d.downCount
+	if t > 300 {
+		t = 300
+	}
+	d.OfflineUntil = time.Now().Add(time.Duration(t) * time.Second)
+	d.downCount += 1
+
+	log.Printf("data node %s is offline for %ds", d.URL.String(), t)
```

Comment:
I would tweak this log message, so it's crystal clear to operators what has happened. I suggest "marking data node %s as offline for...."